fix(VRating): add icon aria-label

damienrobinson opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://vuetifyjs.com/getting-started/contributing

Testing and markup sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->

<!-- We use conventional-changelog-angular for all commit structures -->
<!-- https://vuetifyjs.com/getting-started/contributing#commit-guidelines-w-commitizen -->


## Description
<!-- Describe your changes in detail -->
<!-- Note any issues that are resolved by this PR -->
<!-- e.g. resolves #4213 or fixes #2312 -->
Add aria-label for rating star buttons. Stars - buttons with v-icon class should have aria-label fe. aria-label="rating-star" even if rating is read only.
fixes #11590

## Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Chrome lighthouse accessibility audit fails due to missing aria-labels for rating buttons.
fixes #11590

## How Has This Been Tested?
<!-- All PR's should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | e2e | none -->
lighthouse audit in chrome dev tools

## Markup:
<!-- Information on how to setup your local development environment can be found here: -->
<!-- https://vuetifyjs.com/getting-started/contributing#setup-dev-environment -->

<!-- Paste markup for testing your change --->
<details>

```vue
<template>
<v-rating
      class="d-flex align-center"
      value="4.5"
      background-color="orange lighten-3"
      color="rgba(242,159,5,1)"
      readonly
      half-increments
      round
      dense
     />
</template>
```
</details>

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

KaelWD reviewed (commented):
This isn't useful, it'll just read out "button rating-star" five times.

damienrobinson commented:
> This isn't useful, it'll just read out "button rating-star" five times.

@KaelWD I have updated the aria-label to `Rating Star ${i + 1}`

johnleider reviewed (dismissed):
- The `Rating Star` string needs to go into the locale files
- Documentation needs to be updated
- Formatting onto multiple lines

damienrobinson commented:
- Added `rating.ariaLabel.icon` to the locale files
- Created an example of setting the `icon-aria-label` prop in the docs
- Formatted the icon attrs onto multiple lines
- Updated presets test snap shots to include updated locales

MajesticPotatoe commented on packages/vuetify/src/components/VRating/VRating.ts:
```diff
@@ -219,7 +223,10 @@ export default mixins(
+        attrs: {
+          tabindex: -1,
+          'aria-label': this.$vuetify.lang.t(this.iconAriaLabel, i + 1),
+        }, // TODO: Add a11y support
```
```suggestion
        },
```

damienrobinson replied:
@MajesticPotatoe I have made renamed the prop and added it to api-generator locale

* [x]  rename `icon-aria-label` -> `icon-label`
* [x]  add prop description to api-generator locale

MajesticPotatoe reviewed (changes requested):
- [x] rename `icon-aria-label` -> `icon-label`
- [x] add prop description to api-generator locale

nekosaur commented on packages/docs/src/examples/v-rating/prop-icon-label.vue:
```diff
@@ -0,0 +1,14 @@
+<template>
+  <v-rating
+    v-model="rating"
+    icon-label="custom icon label text"
```
this example does not make use of `{0}`

damienrobinson replied:
I have added {0} and {1} to the example icon-label to now read `custom icon label text {0} of {1}`.

nekosaur commented on packages/vuetify/src/locale/en.ts:
```diff
@@ -61,4 +61,9 @@ export default {
   },
+  rating: {
+    ariaLabel: {
+      icon: 'Rating Star {0}',
```
Not sure if `Star` needs to be part of the text. The actual icon could be anything, not necessarily stars. I also think it would be useful to include the max number, something like `Rating x of y`

damienrobinson commented:
@nekosaur I have removed the text `Star` from the default icon label and added the rating length. The icon label is now `Rating {0} of {1}`.

MajesticPotatoe commented on packages/docs/src/pages/en/components/ratings.md:
```diff
@@ -53,6 +53,12 @@ Utilize the same sizing classes available in `v-icon` or provide your own with t
+
+Provide a label to assistive technologies for icons.
+
+<example file="v-rating/icon-label" />
```
```suggestion
<example file="v-rating/prop-icon-label" />
```

MajesticPotatoe reviewed (approved):
👍 
Thank you for your contribution and interest in improving Vuetify! Make sure to join us in the [Discord community](https://community.vuetifyjs.com).
